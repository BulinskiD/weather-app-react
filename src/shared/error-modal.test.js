import React from 'react';
import ReactDOM from 'react-dom';
import ErrorModal from './error-modal';
import {shallow} from "enzyme";


describe('Error modal', ()=>{
    const clickFn = jest.fn();
    it('should render correctly with given props', ()=>{
        const component = shallow(<ErrorModal onClick={clickFn}>Error</ErrorModal>);
        expect(component).toMatchSnapshot();
    });

    it('should invoke clickFn when button is clicked', ()=>{
        const component = shallow(<ErrorModal onClick={clickFn}>Error</ErrorModal>);
        component.find("Button").simulate('click');

        expect(clickFn).toHaveBeenCalledTimes(1);
    });
});
