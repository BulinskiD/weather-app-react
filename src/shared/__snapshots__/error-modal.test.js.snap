// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error modal should render correctly with given props 1`] = `
<ModalDialog>
  <ModalHeader
    closeButton={false}
    closeLabel="Close"
  >
    <ModalTitle>
      Wystąpił błąd!
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <div
      className="text-center"
    >
      Error
    </div>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      type="button"
      variant="danger"
    >
      Zamknij
    </Button>
  </ModalFooter>
</ModalDialog>
`;
